### PR TITLE
Fix or remove remaining todos

### PR DESCRIPTION
**Pull request by Fraser999**
This PR removes the remaining `todo!` comments from the codebase.  Of these, two remain unresolved but have been ticketed for fixing.

It also fixes an issue in the block synchronizer where an error when applying an `ExecutionResultOrChunk` was being silently ignored.

### CODE REVIEW

File: node/src/components/storage.rs [hunk 1]
```diff
@@ -1269,8 +1269,6 @@ impl Storage {
                 {
                     continue;
                 }
-                // todo!() - right deploy with incorrect approvals found in DB, download and
```
Comment (alsrdn) [inline node/src/components/storage.rs:1311]: Wouldn't we still want to download the correct approvals in this case?

**Review by alsrdn — APPROVED** [review]
LGTM

File: node/src/components/storage.rs (hunk 1 above)
Comment (Fraser999) [reply node/src/components/storage.rs:1311]: I think that's what would happen when the node falls back to leaping and gets this block via historical sync.

**Comment by Fraser999** [comment]
bors merge

**Comment by Fraser999** [comment]
bors merge

**Comment by Fraser999** [comment]
bors merge

**Comment by Fraser999** [comment]
bors try

**Comment by Fraser999** [comment]
bors try

**Comment by Fraser999** [comment]
bors merge

**Pull request merged**